From 0a3b1df764834a78557368c453c86c8499351e9b Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Wed, 2 Mar 2016 16:17:20 +0200
Subject: [PATCH 0911/1240] sar: cmd: add Sample at Reset Chip dump registers

- This command is useful for debugging the system's sample-at-reset
  configuration.
- SYSTEMSW-2311

Change-Id: Ibfda1123fdd262991738aad1bb93bab7688a04bc
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27993
Reviewed-by: Haim Boot <hayim@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
---
 common/mvebu/cmd_sar.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/common/mvebu/cmd_sar.c b/common/mvebu/cmd_sar.c
index 0e40e92..f6af916 100644
--- a/common/mvebu/cmd_sar.c
+++ b/common/mvebu/cmd_sar.c
@@ -23,6 +23,7 @@
 #include <errno.h>
 
 #include "../../board/mvebu/common/sar.h"
+#include "mvebu_chip_sar.h"
 
 int do_sar_cmd(cmd_tbl_t *cmdtp, int flag, int argc,
 			char * const argv[])
@@ -91,3 +92,18 @@ U_BOOT_CMD(
 	"\twrite x y	- Write y to SAR variable x\n"
 	"\tread x	- Read SAR variable x\n"
 );
+
+#ifdef CONFIG_MVEBU_CHIP_SAR
+int do_sar_chip_dump_cmd(cmd_tbl_t *cmdtp, int flag, int argc,
+		char * const argv[])
+{
+	mvebu_sar_dump();
+	return 0;
+}
+
+U_BOOT_CMD(
+	sar_chip_dump,      1,     1,      do_sar_chip_dump_cmd,
+	"sar_chip_dump - dump sample at reset options (from the chip)\n",
+	"\n"
+);
+#endif
-- 
1.9.1

